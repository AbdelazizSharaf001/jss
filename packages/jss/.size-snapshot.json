{
  "dist/jss.js": {
    "bundled": 77413,
    "minified": 21696,
    "gzipped": 6419
  },
  "dist/jss.min.js": {
    "bundled": 77413,
    "minified": 21696,
    "gzipped": 6419
  }
}
