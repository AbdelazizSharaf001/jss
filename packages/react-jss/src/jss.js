// @flow
import {create} from 'jss'
import preset from 'jss-preset-default'

export const jss = create(preset())
