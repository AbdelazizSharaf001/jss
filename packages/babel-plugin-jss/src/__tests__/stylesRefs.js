import {stripIndent} from 'common-tags'
import {sheets} from 'jss'
import {transform} from '../testUtils'

describe('variables refs', () => {
  beforeEach(() => {
    sheets.reset()
  })

  test('support property identifier', () => {
    const code = stripIndent`
      import jss from 'jss';
      const prop = 'a'
      createStyleSheet({
        [prop]: {
          color: 'red'
        }
      });
    `
    expect(transform(code)).toMatchSnapshot()
  })

  test('resolve styles ref', () => {
    const code = stripIndent`
      import jss from 'jss';
      const styles = {
        a: {
          color: 'red'
        }
      };
      createStyleSheet(styles);
    `
    expect(transform(code)).toMatchSnapshot()
  })

  test('extract styles with nested references', () => {
    const code = stripIndent`
      import jss from 'jss';
      const color = 'red';
      const a = {
        color: color
      };
      const styles = {
        a: a
      };
      createStyleSheet(styles);
    `
    expect(transform(code)).toMatchSnapshot()
  })

  test('resolve property access from scope var', () => {
    const code = stripIndent`
      import jss from 'jss';
      const config = {
        primary: 'red'
      };
      createStyleSheet({
        a: {
          color: config.primary
        }
      });
    `
    expect(transform(code)).toMatchSnapshot()
  })
})
