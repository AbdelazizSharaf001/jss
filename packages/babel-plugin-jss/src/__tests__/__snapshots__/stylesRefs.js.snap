// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`variables refs extract styles with nested references 1`] = `
"[v6]
import jss from 'jss';
const color = 'red';
const a = {
  color: color
};
const styles = {
  '@raw': '.a-id {\\\\n  color: red;\\\\n}'
};
createStyleSheet(styles, {
  'classes': {
    'a': 'a-id'
  }
});
[v7]
import jss from 'jss';
const color = 'red';
const a = {
  color: color
};
const styles = {
  \\"@raw\\": \\".a-id {\\\\n  color: red;\\\\n}\\"
};
createStyleSheet(styles, {
  \\"classes\\": {
    \\"a\\": \\"a-id\\"
  }
});"
`;

exports[`variables refs resolve property access from scope var 1`] = `
"[v6]
import jss from 'jss';
const config = {
  primary: 'red'
};
createStyleSheet({
  '@raw': '.a-id {\\\\n  color: red;\\\\n}'
}, {
  'classes': {
    'a': 'a-id'
  }
});
[v7]
import jss from 'jss';
const config = {
  primary: 'red'
};
createStyleSheet({
  \\"@raw\\": \\".a-id {\\\\n  color: red;\\\\n}\\"
}, {
  \\"classes\\": {
    \\"a\\": \\"a-id\\"
  }
});"
`;

exports[`variables refs resolve styles ref 1`] = `
"[v6]
import jss from 'jss';
const styles = {
  '@raw': '.a-id {\\\\n  color: red;\\\\n}'
};
createStyleSheet(styles, {
  'classes': {
    'a': 'a-id'
  }
});
[v7]
import jss from 'jss';
const styles = {
  \\"@raw\\": \\".a-id {\\\\n  color: red;\\\\n}\\"
};
createStyleSheet(styles, {
  \\"classes\\": {
    \\"a\\": \\"a-id\\"
  }
});"
`;

exports[`variables refs support property identifier 1`] = `
"[v6]
import jss from 'jss';
const prop = 'a';
createStyleSheet({
  '@raw': '.a-id {\\\\n  color: red;\\\\n}'
}, {
  'classes': {
    'a': 'a-id'
  }
});
[v7]
import jss from 'jss';
const prop = 'a';
createStyleSheet({
  \\"@raw\\": \\".a-id {\\\\n  color: red;\\\\n}\\"
}, {
  \\"classes\\": {
    \\"a\\": \\"a-id\\"
  }
});"
`;
